File: html-report/src/components/Pool.js
import React, {Component} from 'react';
import cx from 'classnames';
import randomColor from 'randomcolor';
import convertTime from './../utils/convertTime';
import paths from './../utils/paths';
import SearchBar from './SearchBar';

export default class Pool extends Component {
  state = {
    colors: null,
    tests: window.pool.tests
  };

  componentWillMount() {
    this.setDevicesLabelColors();
    document.title = `Pool ${window.pool.id}`;
  }

  setDevicesLabelColors() {
    const generatedColors = randomColor({
      count: window.pool.devices.length,
      luminosity: 'bright'
    });
    let colors = {};
    window.pool.devices.map((item, i) => {
      colors[item.id] = generatedColors[i];
    });
    this.setState({ colors });
  }

  getSearchResults(results) {
    this.setState({ tests: results });
  }

  render() {
    const data = window.pool;
    return (
      <div className="content margin-top-20">
        <div className="title-common"><a href={ paths.fromPoolToIndex }>Pools list</a>/ Pool {data.id}</div>

        <SearchBar setSearchResults={ (results) => this.getSearchResults(results) } />

        <div className="card">
          <div className="vertical-aligned-content title-common">
            <div className="margin-right-10">Tests</div>
            <span className="label">{ this.state.tests.length }</span>
          </div>
          <div className="container-expanded list">
            { this.state.tests.map((test, i) => {
                return (<a key={i} href={`${data.id}/${test.deviceId}/${test.filename}`}
                           className={cx('list__item', 'row full justify-between', test.status)}>
                    <div>
                        <div className="margin-bottom-5 text-sub-title">{test.name}</div>
                        <div className="title-l text-sub-title margin-bottom-5 margin-right-10">{test.class_name}</div>
                        <div className="margin-bottom-5">{test.package_name}</div>
                    </div>
                    <div className="labels-list">
                        <div className="margin-bottom-5">
                    <span className="label info"
                          style={{background: this.state.colors[test.deviceId]}}>{test.deviceId}</span>
                        </div>
                        <div className="margin-bottom-5">
                            <span className="label big">{convertTime(test.duration_millis)}</span>
                        </div>
                    </div>
                </a>)
            }) }
          </div>
        </div>
      </div>
    );
  }
}

File: report/marathon/src/main/kotlin/marathon/component/pool/Pool.kt
@file:OptIn(ExperimentalTime::class)

package marathon.component.pool

import kotlinx.browser.document
import marathon.extensions.format
import react.RBuilder
import react.RComponent
import react.State
import react.dom.html.ReactHTML.a
import react.dom.html.ReactHTML.div
import react.dom.html.ReactHTML.span
import react.router.dom.NavLink
import kotlin.time.Duration
import kotlin.time.ExperimentalTime

class Pool(props: PoolProps) : RComponent<PoolProps, State>(props) {

    override fun componentWillMount() {
        super.componentWillMount()
        document.title = "Pool ${props.pool.id}"
    }

    override fun RBuilder.render() {
        div {
            attrs.className = "content margin-top-20"
            div {
                attrs.className = "title-common"

                a {
                    attrs.href = "toIndex"
                    +"Pools list"
                }

                +"Pool ${props.pool.id}"
            }

            div {
                attrs.className = "card"

                div {
                    attrs.className = "vertical-aligned-content title-common"
                    div {
                        attrs.className = "margin-right-10"
                        +"Tests"
                    }
                    span {
                        attrs.className = "label"
                        +props.pool.tests.size
                    }
                }

                div {
                    attrs.className = "container-expanded list"
                    props.pool.tests.forEachIndexed { index, test ->
                        NavLink {
                            key = index.toString()
                            attrs {
                                href = "/test"
                                className = "list__item row full justify-between ${test.status}"
                            }

                            div {
                                div {
                                    attrs.className = "margin-bottom-5 text-sub-title"
                                    +test.name
                                }
                                div {
                                    attrs.className = "title-l text-sub-title margin-bottom-5 margin-right-10"
                                    +test.class_name
                                }
                                div {
                                    attrs.className = "margin-bottom-5"
                                    +test.package_name
                                }
                            }

                            div {
                                attrs.className = "labels-list"
                                div {
                                    attrs.className = "margin-bottom-5"
                                    span {
                                        attrs.className = "label info"
                                        +test.deviceId
                                        //TODO: colorize the device
                                    }
                                }
                                div {
                                    attrs.className = "margin-bottom-5"
                                    span {
                                        attrs.className = "label big"
                                        +Duration.milliseconds(test.duration_millis.toLong()).format()
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
